test/verify/test_nonstd_gather.cpp

```diff
@@ -0,0 +1,26 @@
+
+#include "verify_program.hpp"
+#include <migraphx/program.hpp>
+#include <migraphx/generate.hpp>
+#include <migraphx/make_op.hpp>
+
+struct test_nonstd_gather : verify_program<test_nonstd_gather>
+{
+    migraphx::program create_program() const
+    {
+        migraphx::program p;
+        auto* mm = p.get_main_module();
+        migraphx::shape s{migraphx::shape::float_type, {3, 3}};
+        migraphx::shape s_indices{migraphx::shape::int32_type, {2, 2}};
+        std::vector<int> indices{1, 1, 0, 2};
+        auto d  = mm->add_parameter("data", s);
+        auto td = mm->add_instruction(migraphx::make_op("transpose", {{"permutation", {1, 0}}}), d);
```

Comment:
Wouldn't auto_contiguous make it standard shape for the gather again?

Reply:
This is true the reference implementation runs auto_contiguous so non-standard shape is not really tested. We should probably have a seperate unit test running `eval` without compilng and compare the results using standard shape input and non-standard shape inputs(they should be the same result).

Reply:
I checked for the CPU as well. It is also using standard shape. 

```
 983 Pass: dead_code_elimination
 984 module: "main"
 985 main:@0 = cpu::preallocate[shape=float_type, {21}, {1},id=main:scratch] -> float_type, {21}, {1}
 986 main:@1 = cpu::literal -> float_type, {2, 2}, {2, 1}
 987 data = @param:data -> float_type, {3, 3}, {3, 1}
 988 main:@2 = transpose[permutation={1, 0}](data) -> float_type, {3, 3}, {1, 3}
 989 main:@3 = load[offset=0,end=36](main:@0) -> float_type, {3, 3}, {3, 1}
 990 main:@4 = dnnl::reorder[post_ops={}](main:@2,main:@3) -> float_type, {3, 3}, {3, 1}
 991 main:@5 = load[offset=36,end=84](main:@0) -> float_type, {3, 2, 2}, {4, 2, 1}
 992 main:@6 = cpu::gather[axis=1](main:@4,main:@1,main:@5) -> float_type, {3, 2, 2}, {4, 2, 1}
 993 main:@7 = @return(main:@6)

```

Reply:
Because `cpu::gather` requires standard shape.

Reply:
@pfultz2 I removed the compile() function for the unit test I added for the gather operator, so input to the gather operator is non-standard. 

Actually, I am wondering whether we should add a list of unit tests for all operators that contain the compute method without compiling the program to see whether they work correctly.